src/nvim/msgpack_rpc/channel.c

```diff
@@ -442,17 +442,22 @@ static void handle_request(Channel *channel, msgpack_object *request)
 
   if (error.set) {
     // Validation failed, send response with error
-    channel_write(channel,
-                  serialize_response(channel->id,
-                                     request_id,
-                                     &error,
-                                     NIL,
-                                     &out_buffer));
-    char buf[256];
-    snprintf(buf, sizeof(buf),
-             "Channel %" PRIu64 " sent an invalid message, closing.",
-             channel->id);
-    call_set_error(channel, buf);
+    bool success = channel_write(channel,
+                                 serialize_response(channel->id,
+                                                    request_id,
+                                                    &error,
+                                                    NIL,
+                                                    &out_buffer));
+
+    // Checking the result of the above write because the channel may
+    // be closed and freed.
```

Comment:
This is a somewhat redundant comment given that `if (success)` is pretty self-documenting. In most of the source, the pattern is usualy `if(channel_write(...))`, so the presence of `success` here is a luxury already. I think the comment can be removed, or at least trimmed to 1 line.